text/XXX-A-Compilation-of-Short-Interactive-Tutorials-and-How-To-Guides-for-Coq.md

```diff
@@ -0,0 +1,257 @@
+
+Once written, we expect the documentation to be relatively stable and easy to maintain as the tutorials and how-to guides should be relatively short, independent, and in limited numbers.
+
+We identify three main reasons to modify an existing tutorial:
+1. Some inaccuracies or omissions have been found by users, the text needs to be clarified, or missing parts need to be added.
+2. We need the tutorial or how-to guide to follow the evolution of best practices.
+3. We need the tutorial or how-to guide to evolve because of new functionalities, incompatibilities or API changes in a new version of the software component.
+
+We plan to use three development branches in order to cope with these modifications:
+- a stable branch, for the current version of the Coq Platform, where changes in the text would be necessary due to the first item above,
+- a main branch, to prepare for the next Coq Platform release with changes mostly for the second and third reasons above,
+- an edge branch, to be compatible with the development versions of Coq and its libraries and plugins.
+
+We would like our edge branch to be part of Coq's CI, and to encourage plugins and library maintainers to integrate the (edge branch) tutorials about their component in their own CI. That way, they can detect incompatibilities in their new versions and correct the tutorials accordingly. These tutorials would also make good test files for their projects.
+
+Furthermore, we expect Coq or Coq plugin and library developers to update tutorials (or at least find someone to do it) when they add a new feature to Coq or a new functionality to their plugins, just as they are supposed to update the reference manual.
+
+
+### 4.4 Online Interface
+
+We have identified 3 different components which play a role in making nice interactive web tutorials:
+- First, we need a tool to format structured Coq comments (coqdoc, embedded `rst`, ...) into `html`. This tool and the associated format for comments should be easy to use to encourage more people to contribute.
+- Second, we need a way to make Coq run inside a standard compliant web browser.
+- Finally, we need a nice user interface inside the interactive web pages which will contain different views: the proof state, Coq's messages, possibly a Search view, ...
+
+We already have some technical answers:
+- [Alectryon](https://github.com/cpitclaudel/alectryon) provides means to interpret restructured text inside Coq files (and conversely).
+- `jsCoq` and `waCoq` enable Coq to be executed inside a web browser, so there is a priori no insurmountable difficulty there.
+
+Yet, it will require some real technical work:
+- waCoq or jsCoq need to evolve to more recent versions of Coq and their usage with external libraries needs to be smoother,
+- we have not tested yet how to use Alectryon and jsCoq/waCoq together.
```

Comment:
Will tutorial writing commence immediately or are these technical changes are necessary prerequisite?  Are somewhat independent tasks?

Do you plan to use any of the Sphinx infrastructure used to generate the reference manual?  It would be good if you could build on that rather than inventing yet another documentation format.

Reply:
Alextryon should be able to convert coqdoc automatically to ReStructured text, should it be retrocompatible. The rest is just UI so it does not matter and we can progress on it independently from the rest.

For Sphinx, maybe @Zimmi48 has an idea ?